soc/arc/snps_qemu/CMakeLists.txt

```diff
@@ -1,5 +1,18 @@
 # SPDX-License-Identifier: Apache-2.0
 
-if(CONFIG_ISA_ARCV2)
-  zephyr_compile_options(-mno-sdata)
+if(COMPILER STREQUAL gcc)
+	zephyr_compile_options(-mcpu=${GCC_M_CPU})
+
+
+	if(CONFIG_ISA_ARCV2)
+	  zephyr_compile_options(-mno-sdata)
+	endif()
+else()
+
+	zephyr_compile_options( -arcv2hs -core2 -Xatomic
```

Comment:
`-arcv2hs` and `-Xatomic` don't seem right for arc_qemu_em - I think some more "if's" will be needed, see e.g. emsdp.

Reply:
Fixed